src/thriveopedia/fossilisation/MuseumCard.cs

```diff
@@ -88,19 +107,52 @@ private void UpdatePreviewImage()
 
     private void OnPressed()
     {
+        // Make sure clicking doesn't pass through from the delete button
+        if (deleteButton?.Pressed ?? false)
+        {
+            Pressed = wasPressed;
+            return;
+        }
+
+        wasPressed = Pressed;
+
         GUICommon.Instance.PlayButtonPressSound();
         EmitSignal(nameof(OnSpeciesSelected), this);
     }
 
     private void OnMouseEnter()
     {
+        if (deleteButton == null)
+            throw new SceneTreeAttachRequired();
+
+        deleteButton.Visible = true;
+
         GUICommon.Instance.Tween.InterpolateProperty(speciesPreview, "modulate", null, Colors.Gray, 0.5f);
         GUICommon.Instance.Tween.Start();
     }
 
     private void OnMouseExit()
     {
+        if (deleteButton == null)
+            throw new SceneTreeAttachRequired();
+
+        deleteButton.Visible = false;
+
         GUICommon.Instance.Tween.InterpolateProperty(speciesPreview, "modulate", null, Colors.White, 0.5f);
         GUICommon.Instance.Tween.Start();
     }
+
+    private void OnDeletePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        deleteConfirmationDialog.PopupCenteredShrink();
+    }
+
+    private void OnDeleteConfirmPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
```

Comment:
I just remembered now that the CustomConfirmationDialog's confirm button already plays its own sound when pressed... Sorry for the mistake but this call should no longer be necessary.

Besides this minor thing, everything looks good to me.

Reply:
Fixed